Providers/Scripts/2.4x-2.5x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,295 @@
+
+
+def check_all_files(src, dest):
+    try:
+        src_files = os.listdir(src)
+        for file_name in src_files:
+            full_src_file = os.path.join(src, file_name)
+            full_dest_file = os.path.join(dest, file_name)
+            if os.path.isfile(full_dest_file):
+                if CompareFiles(full_dest_file, full_src_file, "md5") == -1:
+                    return False
+            else:
+                return False
+        return True
+    except:
+        LG().Log('ERROR', 'check_all_files failed for src: ' + src + ' dest: ' + dest)
+        return False
+
+def CompareFiles(DestinationPath, SourcePath, Checksum):
+    """
+    If the files differ in size, return -1.
+    Reading and computing the hash here is done in a block-by-block manner,
+    in case the file is quite large.
+    """
+    if SourcePath == DestinationPath:  # Files are the same!
+        return 0
+    stat_dest = StatFile(DestinationPath)
+    stat_src = StatFile(SourcePath)
+
+    if stat_src.st_size != stat_dest.st_size:
+        return -1
+
+    if Checksum == "md5":
+        src_error = None
+        dest_error = None
+        src_hash = md5const()
+        dest_hash = md5const()
+        src_block = 'loopme'
+        dest_block = 'loopme'
+        src_file,src_error = opened_bin_w_error(SourcePath, 'rb')
+        if src_error:
+            Print("Exception opening source file " + SourcePath + " Error : " + str(src_error), file=sys.stderr)
+            LG().Log('ERROR', "Exception opening source file " + SourcePath + " Error : " + str(src_error))
+            return -1
+        dest_file, dest_error = opened_bin_w_error(DestinationPath, 'rb')
+        if dest_error:
+            Print("Exception opening destination file " + DestinationPath + " Error : " + str(dest_error), file=sys.stderr)
+            LG().Log('ERROR', "Exception opening destination file " + DestinationPath + " Error : " + str(dest_error))
+            src_file.close()
+            return -1
+        while src_block != '' and dest_block != '':
+            src_block = src_file.read(BLOCK_SIZE)
+            dest_block = dest_file.read(BLOCK_SIZE)
+            src_hash.update(src_block)
+            dest_hash.update(dest_block)
+            if src_hash.hexdigest() != dest_hash.hexdigest():
+                src_file.close()
+                dest_file.close()
+                return -1  
+        if src_hash.hexdigest() == dest_hash.hexdigest():
+            src_file.close()
+            dest_file.close()
+            return 0  
+    elif Checksum == "ctime":
+        if stat_src.st_ctime != stat_dest.st_ctime:
+            return -1
+        else:
+            return 0
+    elif Checksum == "mtime":
+        if stat_src.st_mtime != stat_dest.st_mtime:
+            return -1
+        else:
+            return 0
+
+def GetUID(User):
+    uid = None
+    try:
+        uid = pwd.getpwnam(User)[2]
+    except KeyError:
+        Print('ERROR: Unknown UID for ' + User, file=sys.stderr)
```

Comment:
I don't believe you need to both print and log this error, but I'm unsure why that standard is used for messaging in some providers (like nxScript).